opal/mca/btl/openib/btl_openib_mca.c

```diff
@@ -641,6 +641,9 @@ int btl_openib_register_mca_params(void)
         /* Don't try to recover from this */
         return OPAL_ERR_OUT_OF_RESOURCE;
     }
+    if (NULL != mca_btl_openib_component.default_recv_qps) {
+        free(mca_btl_openib_component.default_recv_qps);
+    }
     mca_btl_openib_component.default_recv_qps = default_qps;
```

Comment:
I would rather save the function call if there is no need for it and if the call is in the critical path. Otherwise, exert your freedom of choice.